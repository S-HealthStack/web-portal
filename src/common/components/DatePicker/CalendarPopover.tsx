import React, { FC, useEffect, useMemo, useState } from 'react';
import _range from 'lodash/range';
import { DateTime } from 'luxon';

import styled from 'styled-components';

import ChevronLeftIcon from 'src/assets/icons/calendar_chevron_left.svg';
import ChevronRightIcon from 'src/assets/icons/calendar_chevron_right.svg';
import { ExtendProps } from 'src/common/utils/types';
import Ripple, { useRipple } from 'src/common/components/Ripple';
import { animation, colors, px, typography } from 'src/styles';

const Container = styled.div`
  width: ${px(312)};
  height: ${px(336)};
  padding: ${px(24)};
  background: ${colors.backgroundSurface};
  border-radius: ${px(4)};
  box-shadow: ${px(0)} ${px(5)} ${px(12)} rgba(0, 0, 0, 0.12);
  display: flex;
  flex-direction: column;
  z-index: 1;
  :hover {
    cursor: default;
  }
`;

const NavigationHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  svg {
    fill: ${colors.textPrimary};
  }
`;

const ControlButtonContainer = styled.button<ControlButtonProps>`
  border: none;
  background-color: transparent;
  height: ${px(24)};
  width: ${px(24)};
  flex: ${px(24)} 0 auto;
  padding: 0;
  margin: 0 ${px(8)};
  border-radius: 50%;
  position: relative;
  z-index: 0;
  transition: background-color 300ms ${animation.defaultTiming};

  &:last-child {
    margin-right: 0;
  }

  &:enabled:hover {
    background-color: ${colors.black08};
    cursor: pointer;
  }

  svg {
    fill: ${colors.textPrimary};
    transition: fill 300ms ${animation.defaultTiming};
  }

  &:disabled {
    svg {
      fill: ${colors.disabled};
    }
  }
`;

const ControlButtonContent = styled.span`
  position: relative;
  z-index: 2;
  display: block;
`;

type ControlButtonProps = React.PropsWithChildren<{ active?: boolean }> &
  React.ButtonHTMLAttributes<HTMLButtonElement>;

// eslint-disable-next-line react/prop-types
const ControlButton: FC<ControlButtonProps> = ({ children, disabled, ...props }) => {
  const { rippleProps, addRippleTriggerProps, handleRippleOut } = useRipple<
    HTMLButtonElement,
    ControlButtonProps
  >({
    color: 'primary',
  });

  useEffect(() => {
    if (disabled) {
      handleRippleOut(undefined, { force: true });
    }
  }, [disabled, handleRippleOut]);

  return (
    <ControlButtonContainer {...addRippleTriggerProps(props)} disabled={disabled}>
      <ControlButtonContent>{children}</ControlButtonContent>
      <Ripple {...rippleProps} />
    </ControlButtonContainer>
  );
};

const NavigationButton = styled(ControlButton)`
  margin: 0;
`;

const HeaderText = styled.div`
  color: ${colors.textPrimary};
  ${typography.bodyMediumSemibold};
`;

const NUM_WEEKS_DISPLAYED = 6;
const NUM_DAYS_PER_WEEK = 7;

const CalendarContainer = styled.div`
  margin-top: ${px(16)};
  display: grid;
  grid-template-columns: repeat(${NUM_DAYS_PER_WEEK}, ${px(36)});
  grid-template-rows: ${px(32)} repeat(${NUM_WEEKS_DISPLAYED}, ${px(36)});
  column-gap: ${px(2)};
  align-self: center;
`;

const CalendarItem = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const WeekdayItem = styled(CalendarItem)`
  ${typography.bodyXSmallRegular};
  color: ${colors.textSecondaryGray};
`;

const DateItem = styled(CalendarItem)<{
  $isDisabled: boolean;
  $isOtherMonth: boolean;
  $isSelected: boolean;
}>`
  ${typography.bodyMediumRegular};
  border-radius: ${px(2)};
  color: ${({ theme, $isSelected, $isDisabled, $isOtherMonth }) =>
    ($isSelected && theme.colors.primaryWhite) ||
    ($isDisabled && theme.colors.disabled) ||
    ($isOtherMonth && theme.colors.textSecondaryGray) ||
    theme.colors.textPrimary};
  background-color: ${({ theme, $isSelected }) => $isSelected && theme.colors.primary};
  &:hover {
    background-color: ${({ theme, $isSelected, $isDisabled }) =>
      !$isSelected && !$isDisabled && theme.colors.primaryLight};
    cursor: ${({ $isDisabled }) => !$isDisabled && 'pointer'};
  }
`;

type Props = ExtendProps<
  React.HTMLAttributes<HTMLDivElement>,
  {
    selectedDate?: Date;
    onSelectedDateChange?: (d: Date) => void;
    minAllowedDate?: Date;
    maxAllowedDate?: Date;
  }
>;

const weekdayLabels = _range(0, NUM_DAYS_PER_WEEK).map(
  (d) => DateTime.fromObject({ weekday: d + 1 }).weekdayShort
);

const CalendarPopover = React.forwardRef(
  (
    { selectedDate, onSelectedDateChange, minAllowedDate, maxAllowedDate, ...rest }: Props,
    ref: React.ForwardedRef<HTMLDivElement>
  ) => {
    const [currentMonth, setCurrentMonth] = useState(
      DateTime.fromJSDate(selectedDate || new Date())
    );
    const adjustCurrentMonth = (change: number) =>
      setCurrentMonth(currentMonth.plus({ months: change }));

    const dates = useMemo(() => {
      const startDate = currentMonth.startOf('month').startOf('week');
      return _range(0, NUM_WEEKS_DISPLAYED * NUM_DAYS_PER_WEEK)
        .map((offsetDays) => startDate.plus({ days: offsetDays }))
        .map((d) => {
          const isCurrentMonth = d.hasSame(currentMonth, 'month');
          const isDisabled =
            (!!minAllowedDate &&
              d.startOf('day') < DateTime.fromJSDate(minAllowedDate).startOf('day')) ||
            (!!maxAllowedDate &&
              d.startOf('day') > DateTime.fromJSDate(maxAllowedDate).startOf('day'));

          return (
            <DateItem
              key={d.toISODate()}
              $isDisabled={isDisabled}
              $isOtherMonth={!isCurrentMonth}
              $isSelected={!!selectedDate && DateTime.fromJSDate(selectedDate).hasSame(d, 'day')}
              onClick={() => {
                if (isDisabled) {
                  return;
                }

                if (!isCurrentMonth) {
                  setCurrentMonth(d);
                }
                onSelectedDateChange?.(d.toJSDate());
              }}
            >
              {d.day}
            </DateItem>
          );
        });
    }, [currentMonth, selectedDate, onSelectedDateChange, minAllowedDate, maxAllowedDate]);

    return (
      <Container {...rest} ref={ref}>
        <NavigationHeader>
          <NavigationButton onClick={() => adjustCurrentMonth(-1)}>
            <ChevronLeftIcon />
          </NavigationButton>
          <HeaderText>{currentMonth.toFormat('MMMM yyyy')}</HeaderText>
          <NavigationButton onClick={() => adjustCurrentMonth(1)}>
            <ChevronRightIcon />
          </NavigationButton>
        </NavigationHeader>
        <CalendarContainer>
          {weekdayLabels.map((d) => (
            <WeekdayItem key={`weekday-${d}`}>{d}</WeekdayItem>
          ))}
          {dates}
        </CalendarContainer>
      </Container>
    );
  }
);

export default CalendarPopover;
