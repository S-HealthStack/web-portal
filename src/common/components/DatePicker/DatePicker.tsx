import { autoUpdate, offset } from '@floating-ui/react-dom';
import { useDismiss, useFloating, useInteractions } from '@floating-ui/react-dom-interactions';
import React, { useCallback, useEffect, useState } from 'react';
import styled from 'styled-components';

import CalendarIcon from 'src/assets/icons/calendar.svg';
import Portal from 'src/common/components/Portal';
import { format } from 'src/common/utils/datetime';
import { ExtendProps } from 'src/common/utils/types';
import { colors, px, typography } from 'src/styles';
import CalendarPopover from './CalendarPopover';

const InputLabel = styled.div<{ $placeholder: boolean }>`
  ${typography.bodyMediumRegular};
  color: ${({ $placeholder, theme }) =>
    $placeholder ? theme.colors.updTextSecondaryGray : theme.colors.updTextPrimaryDark};
`;

const CalendarIconStyled = styled(CalendarIcon)``;

const InputContainer = styled.div<{ $disabled?: boolean; $highlight?: boolean }>`
  max-width: ${px(312)};
  height: ${px(56)};
  background-color: ${colors.updBackground};
  border-radius: ${px(4)};
  border-width: ${px(1)};
  border-color: ${({ $highlight, theme }) =>
    $highlight ? theme.colors.updPrimary : 'transparent'};
  border-style: solid;

  display: flex;
  align-items: center;
  justify-content: space-between;
  position: relative;

  ${InputLabel} {
    margin-left: ${px(15)};
    color: ${({ $disabled, theme }) =>
      $disabled ? theme.colors.updDisabled : theme.colors.updTextPrimaryDark};
  }

  ${CalendarIconStyled} {
    margin-right: ${px(7)};
    fill: ${({ $disabled, theme }) =>
      $disabled ? theme.colors.updPrimaryDisabled : theme.colors.updPrimary};
  }
`;

type Props = ExtendProps<
  React.HTMLAttributes<HTMLDivElement>,
  {
    value?: Date;
    onChange: (value: Date) => void;
    disabled?: boolean;
    min?: Date;
    max?: Date;
    portal?: boolean;
    updatePosition?: boolean;
  }
>;

const DatePicker: React.FC<Props> = ({
  value,
  onChange,
  disabled,
  min,
  max,
  portal,
  updatePosition,
  ...rest
}) => {
  const [isOpen, setOpen] = useState(false);

  const popover = useFloating({
    middleware: [offset(8)],
    placement: 'bottom',
    open: isOpen,
    onOpenChange: setOpen,
    whileElementsMounted: updatePosition || portal ? autoUpdate : undefined,
  });
  const popoverInteractions = useInteractions([useDismiss(popover.context)]);

  useEffect(() => {
    if (disabled) {
      setOpen(false);
    }
  }, [disabled]);

  const handleSelectedDateChange = useCallback(
    (d: Date) => {
      setOpen(false);
      onChange(d);
    },
    [onChange]
  );

  const popoverElement = isOpen && (
    <CalendarPopover
      {...popoverInteractions.getFloatingProps({
        ref: popover.floating,
        style: {
          position: popover.strategy,
          top: popover.y ?? 0,
          left: popover.x ?? 0,
        },
      })}
      selectedDate={value}
      onSelectedDateChange={handleSelectedDateChange}
      minAllowedDate={min}
      maxAllowedDate={max}
    />
  );

  return (
    <InputContainer
      {...popoverInteractions.getReferenceProps({
        ...rest,
        ref: popover.reference,
        onClick() {
          if (!disabled && !isOpen) {
            setOpen(true);
          }
        },
      })}
      $disabled={disabled}
      $highlight={value && isOpen}
    >
      <InputLabel $placeholder={!value}>
        {value ? format(value, 'EEE, MMM dd, yyyy') : 'Select'}
      </InputLabel>
      <CalendarIconStyled />
      {portal ? (
        <Portal id="date-picker" enabled={isOpen}>
          {popoverElement}
        </Portal>
      ) : (
        popoverElement
      )}
    </InputContainer>
  );
};

export default DatePicker;
