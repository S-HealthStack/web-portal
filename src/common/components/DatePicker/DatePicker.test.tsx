import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { getByText, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import theme from 'src/styles/theme';
import { ThemeProvider } from 'styled-components/';
import DatePicker from './DatePicker';

describe('DatePicker', () => {
  it('test date picker render', async () => {
    const onChange = jest.fn();
    const testDate = new Date(2022, 0, 1);

    const { baseElement, getByTestId, queryByTestId, rerender } = render(
      <ThemeProvider theme={theme}>
        <DatePicker onChange={onChange} data-testid="date-picker" />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const datePicker = getByTestId('date-picker') as Element;
    expect(datePicker).toBeInTheDocument();

    expect(getByText(baseElement, 'Select')).toBeInTheDocument();

    rerender(
      <ThemeProvider theme={theme}>
        <DatePicker onChange={onChange} value={testDate} data-testid="date-picker" />
      </ThemeProvider>
    );

    await userEvent.click(datePicker);

    await waitFor(() => expect(queryByTestId('calendar-popover')).toBeVisible(), { timeout: 100 });

    expect(baseElement).toMatchSnapshot();

    expect(getByText(baseElement, 'Sat, Jan 01, 2022')).toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });

  it('[NEGATIVE] test date picker render with broken parameters', async () => {
    const onChange = jest.fn();
    const testDate = null;

    const { baseElement, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <DatePicker
          onChange={onChange}
          value={testDate as unknown as Date}
          data-testid="date-picker"
        />
      </ThemeProvider>
    );

    const datePicker = getByTestId('date-picker') as Element;

    expect(baseElement).toMatchSnapshot();
    expect(datePicker).toBeInTheDocument();
  });
});
