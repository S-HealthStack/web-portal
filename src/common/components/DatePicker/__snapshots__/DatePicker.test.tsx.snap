// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker test date picker render 1`] = `
.c2 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
}

.c0 {
  max-width: 19.5rem;
  height: 3.5rem;
  background-color: #F7F8FA;
  border-radius: 0.25rem;
  border-width: 0.0625rem;
  border-color: transparent;
  border-style: solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.c0:hover {
  cursor: pointer;
}

.c0 .c1 {
  margin-left: 0.9375rem;
  color: #474747;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="date-picker"
    >
      <div
        class="c1 c2"
        data-testid="input-label"
      >
        Select
      </div>
      <svg
        class=""
      />
    </div>
  </div>
</body>
`;

exports[`DatePicker test date picker render 2`] = `
.c8 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #4475E3;
  pointer-events: none;
}

.c3 {
  width: 19.5rem;
  height: 21rem;
  padding: 1.5rem;
  background: #FFFFFF;
  border-radius: 0.25rem;
  box-shadow: 0rem 0.3125rem 0.75rem rgba(0,0,0,0.12);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  z-index: 1;
}

.c3:hover {
  cursor: default;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 svg {
  fill: #474747;
}

.c5 {
  border: none;
  background-color: transparent;
  height: 1.5rem;
  width: 1.5rem;
  -webkit-flex: 1.5rem 0 auto;
  -ms-flex: 1.5rem 0 auto;
  flex: 1.5rem 0 auto;
  padding: 0;
  margin: 0 0.5rem;
  border-radius: 50%;
  position: relative;
  z-index: 0;
  -webkit-transition: background-color 300ms cubic-bezier(0.45,0.05,0,1);
  transition: background-color 300ms cubic-bezier(0.45,0.05,0,1);
}

.c5:last-child {
  margin-right: 0;
}

.c5:enabled:hover {
  background-color: rgba(0,0,0,0.08);
  cursor: pointer;
}

.c5 svg {
  fill: #474747;
  -webkit-transition: fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: fill 300ms cubic-bezier(0.45,0.05,0,1);
}

.c5:disabled svg {
  fill: #DBDBDB;
}

.c7 {
  position: relative;
  z-index: 2;
  display: block;
}

.c6 {
  margin: 0;
}

.c9 {
  color: #474747;
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c10 {
  margin-top: 1rem;
  display: grid;
  grid-template-columns: repeat(7,2.25rem);
  grid-template-rows: 2rem repeat(6,2.25rem);
  -webkit-column-gap: 0.125rem;
  column-gap: 0.125rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c12 {
  font-family: Inter,sans-serif;
  font-size: 0.75rem;
  line-height: 1.125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
}

.c13 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  border-radius: 0.125rem;
  color: #808080;
}

.c13:hover {
  background-color: #ECF1FC;
  cursor: pointer;
}

.c14 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  border-radius: 0.125rem;
  color: #FFFFFF;
  background-color: #4475E3;
}

.c14:hover {
  cursor: pointer;
}

.c15 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  border-radius: 0.125rem;
  color: #474747;
}

.c15:hover {
  background-color: #ECF1FC;
  cursor: pointer;
}

.c2 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
}

.c16 .c1 {
  margin-left: 0.9375rem;
  color: #474747;
}

.c0 {
  max-width: 19.5rem;
  height: 3.5rem;
  background-color: #F7F8FA;
  border-radius: 0.25rem;
  border-width: 0.0625rem;
  border-color: #4475E3;
  border-style: solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.c0:hover {
  cursor: pointer;
}

.c0 .c1 {
  margin-left: 0.9375rem;
  color: #474747;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="date-picker"
    >
      <div
        class="c1 c2"
        data-testid="input-label"
      >
        Sat, Jan 01, 2022
      </div>
      <svg
        class=""
      />
      <div
        class="c3"
        data-testid="calendar-popover"
        style="position: absolute; top: 8px; left: 0px;"
        tabindex="-1"
      >
        <div
          class="c4"
        >
          <button
            class="c5 c6"
            style="overflow: hidden; position: relative;"
          >
            <span
              class="c7"
            >
              <svg />
            </span>
            <div
              class="c8"
              color="primary"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
          <div
            class="c9"
          >
            January 2022
          </div>
          <button
            class="c5 c6"
            style="overflow: hidden; position: relative;"
          >
            <span
              class="c7"
            >
              <svg />
            </span>
            <div
              class="c8"
              color="primary"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
        </div>
        <div
          class="c10"
        >
          <div
            class="c11 c12"
          >
            Mon
          </div>
          <div
            class="c11 c12"
          >
            Tue
          </div>
          <div
            class="c11 c12"
          >
            Wed
          </div>
          <div
            class="c11 c12"
          >
            Thu
          </div>
          <div
            class="c11 c12"
          >
            Fri
          </div>
          <div
            class="c11 c12"
          >
            Sat
          </div>
          <div
            class="c11 c12"
          >
            Sun
          </div>
          <div
            class="c11 c13"
          >
            27
          </div>
          <div
            class="c11 c13"
          >
            28
          </div>
          <div
            class="c11 c13"
          >
            29
          </div>
          <div
            class="c11 c13"
          >
            30
          </div>
          <div
            class="c11 c13"
          >
            31
          </div>
          <div
            class="c11 c14"
          >
            1
          </div>
          <div
            class="c11 c15"
          >
            2
          </div>
          <div
            class="c11 c15"
          >
            3
          </div>
          <div
            class="c11 c15"
          >
            4
          </div>
          <div
            class="c11 c15"
          >
            5
          </div>
          <div
            class="c11 c15"
          >
            6
          </div>
          <div
            class="c11 c15"
          >
            7
          </div>
          <div
            class="c11 c15"
          >
            8
          </div>
          <div
            class="c11 c15"
          >
            9
          </div>
          <div
            class="c11 c15"
          >
            10
          </div>
          <div
            class="c11 c15"
          >
            11
          </div>
          <div
            class="c11 c15"
          >
            12
          </div>
          <div
            class="c11 c15"
          >
            13
          </div>
          <div
            class="c11 c15"
          >
            14
          </div>
          <div
            class="c11 c15"
          >
            15
          </div>
          <div
            class="c11 c15"
          >
            16
          </div>
          <div
            class="c11 c15"
          >
            17
          </div>
          <div
            class="c11 c15"
          >
            18
          </div>
          <div
            class="c11 c15"
          >
            19
          </div>
          <div
            class="c11 c15"
          >
            20
          </div>
          <div
            class="c11 c15"
          >
            21
          </div>
          <div
            class="c11 c15"
          >
            22
          </div>
          <div
            class="c11 c15"
          >
            23
          </div>
          <div
            class="c11 c15"
          >
            24
          </div>
          <div
            class="c11 c15"
          >
            25
          </div>
          <div
            class="c11 c15"
          >
            26
          </div>
          <div
            class="c11 c15"
          >
            27
          </div>
          <div
            class="c11 c15"
          >
            28
          </div>
          <div
            class="c11 c15"
          >
            29
          </div>
          <div
            class="c11 c15"
          >
            30
          </div>
          <div
            class="c11 c15"
          >
            31
          </div>
          <div
            class="c11 c13"
          >
            1
          </div>
          <div
            class="c11 c13"
          >
            2
          </div>
          <div
            class="c11 c13"
          >
            3
          </div>
          <div
            class="c11 c13"
          >
            4
          </div>
          <div
            class="c11 c13"
          >
            5
          </div>
          <div
            class="c11 c13"
          >
            6
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`DatePicker test date picker render 3`] = `
.c8 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #4475E3;
  pointer-events: none;
}

.c3 {
  width: 19.5rem;
  height: 21rem;
  padding: 1.5rem;
  background: #FFFFFF;
  border-radius: 0.25rem;
  box-shadow: 0rem 0.3125rem 0.75rem rgba(0,0,0,0.12);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  z-index: 1;
}

.c3:hover {
  cursor: default;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 svg {
  fill: #474747;
}

.c5 {
  border: none;
  background-color: transparent;
  height: 1.5rem;
  width: 1.5rem;
  -webkit-flex: 1.5rem 0 auto;
  -ms-flex: 1.5rem 0 auto;
  flex: 1.5rem 0 auto;
  padding: 0;
  margin: 0 0.5rem;
  border-radius: 50%;
  position: relative;
  z-index: 0;
  -webkit-transition: background-color 300ms cubic-bezier(0.45,0.05,0,1);
  transition: background-color 300ms cubic-bezier(0.45,0.05,0,1);
}

.c5:last-child {
  margin-right: 0;
}

.c5:enabled:hover {
  background-color: rgba(0,0,0,0.08);
  cursor: pointer;
}

.c5 svg {
  fill: #474747;
  -webkit-transition: fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: fill 300ms cubic-bezier(0.45,0.05,0,1);
}

.c5:disabled svg {
  fill: #DBDBDB;
}

.c7 {
  position: relative;
  z-index: 2;
  display: block;
}

.c6 {
  margin: 0;
}

.c9 {
  color: #474747;
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c10 {
  margin-top: 1rem;
  display: grid;
  grid-template-columns: repeat(7,2.25rem);
  grid-template-rows: 2rem repeat(6,2.25rem);
  -webkit-column-gap: 0.125rem;
  column-gap: 0.125rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c12 {
  font-family: Inter,sans-serif;
  font-size: 0.75rem;
  line-height: 1.125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
}

.c13 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  border-radius: 0.125rem;
  color: #808080;
}

.c13:hover {
  background-color: #ECF1FC;
  cursor: pointer;
}

.c14 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  border-radius: 0.125rem;
  color: #FFFFFF;
  background-color: #4475E3;
}

.c14:hover {
  cursor: pointer;
}

.c15 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  border-radius: 0.125rem;
  color: #474747;
}

.c15:hover {
  background-color: #ECF1FC;
  cursor: pointer;
}

.c2 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
}

.c16 .c1 {
  margin-left: 0.9375rem;
  color: #474747;
}

.c0 {
  max-width: 19.5rem;
  height: 3.5rem;
  background-color: #F7F8FA;
  border-radius: 0.25rem;
  border-width: 0.0625rem;
  border-color: #4475E3;
  border-style: solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.c0:hover {
  cursor: pointer;
}

.c0 .c1 {
  margin-left: 0.9375rem;
  color: #474747;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="date-picker"
    >
      <div
        class="c1 c2"
        data-testid="input-label"
      >
        Sat, Jan 01, 2022
      </div>
      <svg
        class=""
      />
      <div
        class="c3"
        data-testid="calendar-popover"
        style="position: absolute; top: 8px; left: 0px;"
        tabindex="-1"
      >
        <div
          class="c4"
        >
          <button
            class="c5 c6"
            style="overflow: hidden; position: relative;"
          >
            <span
              class="c7"
            >
              <svg />
            </span>
            <div
              class="c8"
              color="primary"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
          <div
            class="c9"
          >
            January 2022
          </div>
          <button
            class="c5 c6"
            style="overflow: hidden; position: relative;"
          >
            <span
              class="c7"
            >
              <svg />
            </span>
            <div
              class="c8"
              color="primary"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
        </div>
        <div
          class="c10"
        >
          <div
            class="c11 c12"
          >
            Mon
          </div>
          <div
            class="c11 c12"
          >
            Tue
          </div>
          <div
            class="c11 c12"
          >
            Wed
          </div>
          <div
            class="c11 c12"
          >
            Thu
          </div>
          <div
            class="c11 c12"
          >
            Fri
          </div>
          <div
            class="c11 c12"
          >
            Sat
          </div>
          <div
            class="c11 c12"
          >
            Sun
          </div>
          <div
            class="c11 c13"
          >
            27
          </div>
          <div
            class="c11 c13"
          >
            28
          </div>
          <div
            class="c11 c13"
          >
            29
          </div>
          <div
            class="c11 c13"
          >
            30
          </div>
          <div
            class="c11 c13"
          >
            31
          </div>
          <div
            class="c11 c14"
          >
            1
          </div>
          <div
            class="c11 c15"
          >
            2
          </div>
          <div
            class="c11 c15"
          >
            3
          </div>
          <div
            class="c11 c15"
          >
            4
          </div>
          <div
            class="c11 c15"
          >
            5
          </div>
          <div
            class="c11 c15"
          >
            6
          </div>
          <div
            class="c11 c15"
          >
            7
          </div>
          <div
            class="c11 c15"
          >
            8
          </div>
          <div
            class="c11 c15"
          >
            9
          </div>
          <div
            class="c11 c15"
          >
            10
          </div>
          <div
            class="c11 c15"
          >
            11
          </div>
          <div
            class="c11 c15"
          >
            12
          </div>
          <div
            class="c11 c15"
          >
            13
          </div>
          <div
            class="c11 c15"
          >
            14
          </div>
          <div
            class="c11 c15"
          >
            15
          </div>
          <div
            class="c11 c15"
          >
            16
          </div>
          <div
            class="c11 c15"
          >
            17
          </div>
          <div
            class="c11 c15"
          >
            18
          </div>
          <div
            class="c11 c15"
          >
            19
          </div>
          <div
            class="c11 c15"
          >
            20
          </div>
          <div
            class="c11 c15"
          >
            21
          </div>
          <div
            class="c11 c15"
          >
            22
          </div>
          <div
            class="c11 c15"
          >
            23
          </div>
          <div
            class="c11 c15"
          >
            24
          </div>
          <div
            class="c11 c15"
          >
            25
          </div>
          <div
            class="c11 c15"
          >
            26
          </div>
          <div
            class="c11 c15"
          >
            27
          </div>
          <div
            class="c11 c15"
          >
            28
          </div>
          <div
            class="c11 c15"
          >
            29
          </div>
          <div
            class="c11 c15"
          >
            30
          </div>
          <div
            class="c11 c15"
          >
            31
          </div>
          <div
            class="c11 c13"
          >
            1
          </div>
          <div
            class="c11 c13"
          >
            2
          </div>
          <div
            class="c11 c13"
          >
            3
          </div>
          <div
            class="c11 c13"
          >
            4
          </div>
          <div
            class="c11 c13"
          >
            5
          </div>
          <div
            class="c11 c13"
          >
            6
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
